src/CMakeLists.txt

```diff
@@ -249,9 +247,9 @@ if(ENABLE_GAME)
 		Boost::filesystem
 		Fontconfig::Fontconfig
 	)
+	target_link_libraries(wesnoth SDL2::SDL2)
```

Comment:
I've removed the `if` here and in other places as well
the whole complexity of SDL2 with pc, cmake-config or vcpkg-style-cmake-config should be hidden now